src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorMatcher.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.TagHelpers
+{
+    /// <summary>
+    /// Helper methods related to <see cref="TagHelperDescriptor"/> and related classes.
+    /// </summary>
+    public static class TagHelperDescriptorMatcher
```

Comment:
I'm horrible with names so I'm not one to offer a better one but `TagHelperDescriptorMatcher` doesn't really make sense for what this class does. Makes me think of an equality comparer. Maybe `TagHelperDescriptorUtility` or `TagHelperDescriptorInspector`?